WordPress/Classes/Services/Stories/WPMediaPicker+MediaPicker.swift

```diff
@@ -161,38 +165,52 @@ extension WPMediaAsset {
     func videoURLPublisher() -> AnyPublisher<URL, Error> {
         return videoAssetPublisher().tryMap { asset -> AnyPublisher<URL, Error> in
             let filename = UUID().uuidString
-            let url = try FileManager.default.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: true).appendingPathComponent(filename).appendingPathExtension("mp4")
+            let url = try FileManager.default.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: true).appendingPathComponent(filename)
             let urlAsset = asset as? AVURLAsset
 
             if let assetURL = urlAsset?.url {
+                let existingExtension = assetURL.pathExtension
+                let exportURL = url.appendingPathExtension(existingExtension)
```

Comment:
Uses the Media's original extension instead of hardcoding it.

Reply:
Maybe `url.appendingPathExtension(assetURL.pathExtension)` since we don't use `existingExtension` for anything else?  Up to you.